### PR TITLE
Check favs

### PR DESCRIPTION
**Pull request by SamiFdev**

### CODE REVIEW

File: src/components/Favorites/favorites.module.css [hunk 1]
```diff
@@ -22,3 +22,7 @@
     display: block;
     color: var(--primary-textColor);
 }
+
+.favStar {
+    fill: gold;
```
Comment (arfusop) [inline src/components/Favorites/favorites.module.css:27]: Just so you know, you don't necessarily have to add a class to the SVG elements. They can be targeted by 'svg' in here. And you can use 'fill' or 'color'. Svg works the same as text, color and font size will apply size and colors

File: src/components/Favorites/index.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+import { AiOutlineStar } from "react-icons/ai";
+import { fetchSingleCoin } from "../../features/singleCoinSlice";
+
+function Favorites() {
+    const { data, fetched } = useSelector((state) => state.favorites);
+    const dispatch = useDispatch();
+
+    useEffect(() => {
+        if (!data?.length && !fetched) {
+            dispatch(importSavedFavorites());
```
Comment (arfusop) [inline src/components/Favorites/index.js:18]: Just make sure you're not doing this in MainContent as well. You only need to do it one time. I can't remember if it's in both or not

File: src/components/Favorites/index.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    };
+
+    if (data?.length) {
+        return (
+            <section className={styles.sidePanel}>
+                <h2>Favorites</h2>
+                <ul>
+                    {data.map((coin, index) => (
+                        <li key={index}>
+                            {coin}
```
Comment (arfusop) [inline src/components/Favorites/index.js:40]: dont leave floating text unwrapped. `span` or `p` tags

File: src/components/Favorites/index.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+                        </li>
+                    ))}
+                </ul>
+                <button onClick={clearFavorites}>clear</button>
+            </section>
+        );
+    } else {
+        return (
+            <section className={styles.sidePanel}>
+                <h2>Favorites</h2>
```
Comment (arfusop) [inline src/components/Favorites/index.js:56]: Add a message here under the `h2` to let the user know how they can add a favorite.

File: src/components/Header/Header.module.css [hunk 5]
```diff
@@ -8,3 +8,58 @@
...
+.coinsContainer {
+    overflow: hidden;
+}
+
+img {
+    height: 2rem;
+    width: 2rem;
+}
+
+/* .scrollingInfo {
```
Comment (arfusop) [inline src/components/Header/Header.module.css:21]: removeeeeee comments

File: src/components/MainContent/index.js [hunk 6]
```diff
@@ -20,20 +27,64 @@ function MainContent() {
...
+    const onFavoriteClick = () => {
+        // dispatch addNewFavorite with argument of coin id saved in state in this component - powering the value of the input
+        dispatch(addNewFavorite(searchedCoin));
+    };
+
+    const onRemoveClick = () => {
+        dispatch(removeExistingFavorite(searchedCoin));
+    };
+
+    // TODO
```
Comment (arfusop) [inline src/components/MainContent/index.js:39]: remove comment

File: src/components/MainContent/index.js [hunk 7]
```diff
@@ -2,15 +2,22 @@ import styles from "./mainContent.module.css";
 import { useState } from "react";
 import { useDispatch, useSelector } from "react-redux";
 import { fetchSingleCoin } from "../../features/singleCoinSlice";
+import { importSavedFavorites } from "../../features/favoritesSlice";
+// import { updateFavorites } from "../../features/favoritesSlice";
```
Comment (arfusop) [inline src/components/MainContent/index.js:6]: remove if youre not using

File: src/components/MainContent/index.js [hunk 8]
```diff
@@ -20,20 +27,64 @@ function MainContent() {
...
-            <button onClick={handleSubmit} className={styles.submitButton}>
+            <button
+                disabled={!searchedCoin.length || loading}
+                onClick={handleSubmit}
+                className={styles.submitButton}
+            >
                 Submit
             </button>
+            <div className={styles.singleCoinDiv}>
+                {/* Use redux favorites slice to indicate if there was no data
```
Comment (arfusop) [inline src/components/MainContent/index.js:62]: trash the comment

File: src/components/MainContent/index.js [hunk 9]
```diff
@@ -20,20 +27,64 @@ function MainContent() {
...
+                returned after fetch */}
+                {!data?.id ? (
+                    <div>Search for a valid coin</div>
+                ) : (
+                    <>
+                        {data.id}
```
Comment (arfusop) [inline src/components/MainContent/index.js:68]: span or p tag

File: src/components/MainContent/index.js [hunk 10]
```diff
@@ -20,20 +27,64 @@ function MainContent() {
...
+                        {data.current_price}
```
Comment (arfusop) [inline src/components/MainContent/index.js:69]: span or p tag

File: src/components/MainContent/index.js [hunk 11]
```diff
@@ -20,20 +27,64 @@ function MainContent() {
...
+                            </button>
+                        ) : (
+                            <button
+                                onClick={onFavoriteClick}
+                                className={styles.favoriteButton}
+                            >
+                                <AiOutlineStar />
+                            </button>
+                        )}
+                        {/* check if data.id exists in favorites */}
```
Comment (arfusop) [inline src/components/MainContent/index.js:82]: remove

File: src/features/favoritesSlice.js [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+            const copiedState = [...state.data, action.payload];
+            state.data = copiedState;
+            localStorage.setItem("favoriteCoinIds", copiedState);
+        },
+        removeExistingFavorite: (state, action) => {
+            // expected payload: coin id to remove
+            const newFavoritesArray = [...state.data];
+            const indexToRemove = newFavoritesArray.indexOf(action.payload);
+            newFavoritesArray.splice(indexToRemove, 1);
+            console.log(newFavoritesArray);
```
Comment (arfusop) [inline src/features/favoritesSlice.js:24]: Get the log out

**Review by arfusop — CHANGES_REQUESTED** [review]

File: src/components/Favorites/index.js (hunk 3 above)
Comment (SamiFdev) [reply src/components/Favorites/index.js:40]: this is being addressed in styling branch.

**Pull request merged**